### ISSUE TITLE
10$: config: allow "globals" to be an array

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#809]
```json
{
  "globals": ["goog"]
}
```

This is less flexible than an object, but matches intuition.

**Comment by TomBinford** [comment]
I'd like to claim this one please.

**Comment by martinsilver** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
@TomBinford Sorry, I missed you message. Are you still interested in this task? If not, I am going to let @martinsilver work on it.

**Comment by TomBinford** [comment]
@strager Martin can have it 👍

**Comment by strager** [comment]
@martinsilver The following quick-lint-js.config file is legal today:
```json
{
  "globals": {
    "google": true,
    "$": true,
    "jQuery": true
  }
}
```
This task is about allowing the user to write the following instead, which is **equivalent**:
```json
{
  "globals": [
    "google",
    "$",
    "jQuery"
  ]
}
```
The config file is interpreted as if the array was an object where each array element makes an object property. Each property's key is the array element, and each proeprty's value is `true`.

If an array value is not a string, that is an error which should be reported to the user.

Expectations:
* Tests
* Documentation
